src/Endpoints/class-related-api-proxy.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * Parsely Related REST API Endpoint
+ *
+ * @package Parsely
+ * @since 3.2.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Endpoints;
+
+use Parsely\Parsely;
+use Parsely\RemoteAPI\Proxy;
+use stdClass;
+use WP_Error;
+use WP_REST_Server;
+use WP_REST_Request;
+
+/**
+ * Configure a REST API endpoint for use e.g. by the Related Block.
+ */
+final class Related_API_Proxy {
+	/**
+	 * Parsely instance.
+	 *
+	 * @var Parsely
+	 */
+	private $parsely;
+
+	/**
+	 * Proxy object which does the actual calls to the Parse.ly API.
+	 *
+	 * @var Proxy
+	 */
+	private $proxy;
+
+	/**
+	 * Used to inject dependencies.
+	 *
+	 * @param Parsely $parsely Instance of Parsely class.
+	 * @param Proxy   $proxy   Proxy object which does the actual calls to the Parse.ly API.
+	 */
+	public function __construct( Parsely $parsely, Proxy $proxy ) {
+		$this->parsely = $parsely;
+		$this->proxy   = $proxy;
+	}
+
+	/**
+	 * Entrypoint to register the endpoint and otherwise initialize this class.
+	 *
+	 * @return void
+	 */
+	public function run(): void {
+		if ( ! apply_filters( 'wp_parsely_enable_related_endpoint', false ) ) {
+			return;
+		}
+
+		$get_items_args = array(
+			'query' => array(
+				'default'           => array(),
+				'sanitize_callback' => function ( $query ) {
+					// question: how should we sanitize these?
+					return (array) $query;
+				},
+			),
+		);
+
+		$rest_route_args = array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'permission_callback' ),
+				'args'                => $get_items_args,
+			),
+		);
+
+		register_rest_route( 'wp-parsely/v1', '/related', $rest_route_args );
+	}
+
+	/**
+	 * Determine if the endpoint can be called.
+	 *
+	 * @return bool
+	 */
+	public function permission_callback(): bool {
+		// Unauthenticated.
+		return true;
```

Comment:
Should we at impose a nonce check or any rate limiting?  Any changes can be done in a follow up, but I thought I'd raise it here at least.

Reply:
Probably needs more thought, so would suggest opening an issue to follow-up.